Add functions used to animate the highlight in the banner text

wmde-manicki opened the pull request:
Those function are simply moved over from https://meta.wikimedia.org/wiki/MediaWiki:WMDE_FR2015/Resources/AnimateHighlight.js with cs things applied. Plus I added a possibility to set custom highlight colour as it turned out to be required by one of the banners on wikipedia.de (of course it has been the last one that I had to update)

WMDE-Fisch commented on js/animateHighlight.js:
```diff
@@ -0,0 +1,36 @@
+}
+
+function highlightSpan( id, num, color ) {
+	$( '#' + id + ' span:nth-child(' + num + ')' ).css( 'background-color', color );
```
Why not setting a class instead of setting a color? could be easier to customize via CSS.

wmde-manicki commented:
Okay, so I have rewritten this thing. As this script now does not rely on anything from the old version, I just removed the old history (you can still compare it with https://meta.wikimedia.org/wiki/MediaWiki:WMDE_FR2015/Resources/AnimateHighlight.js)
The most important reason to have different names for all functions is that it will allow to deploy it on Meta and update exisiting CN banners without breaking them for those 15 minutes until CN refreshes the cache.

Changes in this approach:
 - single function call instead of two, ie. only `animateHighlight( $( '#banner-text-highlight' ), 'highlight', 10 );` needed in banner code (plus optional timeout)
- does not require to pass start and end characters (assumes highligh all text in the given elem unless specified differently)
 - highlight is done by adding a class instead of changing the color
 - single character <span>s don't have custom in-line style - It looks to me that custom padding of "character" <span>s has been added only done `btext-highlight` sets the padding too. I'd rather use different class to highlight that only changes colour, and additional padding can than be added with other styles (if wanted), e.g. like below:
```
#to-highlight span { padding: 1px 0; }
#to-highlight span:first-child { padding-left: 5px; }
#to-highlight span:last-child { padding-right: 5px; }
```
 - "tempo" is "speed"
 - get $object instead of an id

I am not totally happy with the way the "animation" is done but given there is no direct way to "sleep" in Javascript, the only other solution I came up with would to be to do N setTimeout calls with increasing time but this is at least as bad as the "old" solution, so I'm not changing that.

Please have another look @KaiNissen @WMDE-Fisch. I'l put it into some banner so it's easier to test.

wmde-manicki commented:
Here: https://de.wikipedia.org/wiki/Liste_der_Staatsoberh%C3%A4upter_637?banner=B15WMDE_151207_fallback
And I think I should put common highlight-related styles to common CSS file (a.k.a. `DesktopBanner.js`)

gbirke commented:
+1

KaiNissen commented:
When the animation starts, the text kind of jumps, but that is a styling issue.

Apart from the (picky) comment I made, this looks good.

wmde-manicki commented:
Thanks @KaiNissen for comments.
Remark about the order of things got me to think that those `<span>s` should only be added for characters that are highlighted (if only part of text is highlighted etc). So I moved counting the number of characters to be done as early as possible, as you suggested, and also changed functions a bit.

Regarding the jumping. I noticed it only once (I believe it is more evident in particular banner text+window width+banner size combinations) but it must have been caused by adding this extra 5 pixels of padding around the highlighted text. It has been only included because the "old version" had it. Personally I don't like it and don't think it is even needed. I removed it then.

Recent changes has been refltected in https://de.wikipedia.org/wiki/Liste_der_Staatsoberh%C3%A4upter_637?banner=B15WMDE_151207_fallback too

gbirke commented:
I talked to Henning about the whitespace before and after the highlighting and we came to the conclusion that leaving out the whitespace is the best option here.

The original effect without the jumping could be achieved with the following CSS:
```CSS
#to-highlight { padding: 0 5px; }
#to-highlight span:first-child { padding-left: 5px; margin-left: -5px; }
#to-highlight span:last-child { padding-right: 5px; margin-right: -5px; }
````
But that leads to unsightly white space (and even more unsightly indentation if the highlighted sentence starts at the beginning of a line). So we just leave it out.

wmde-manicki commented:
Good point. I moved a level higher.
I also renamed the `speed` parameter as I realized it is a bit misleading. I would expect higher "speed" to make the animation faster while the param is actually doing the oposite.

KaiNissen commented:
:+1:
